diff -uprN a/libipvs/libipvs.c b/libipvs/libipvs.c
--- a/libipvs/libipvs.c	2011-02-07 10:38:57.000000000 +0800
+++ b/libipvs/libipvs.c	2017-03-31 15:12:49.991580457 +0800
@@ -32,7 +32,7 @@ static void* ipvs_func = NULL;
 struct ip_vs_getinfo ipvs_info;
 
 #ifdef LIBIPVS_USE_NL
-static struct nl_handle *sock = NULL;
+static struct nl_sock *sock = NULL;
 static int family, try_nl = 1;
 #endif
 
@@ -73,7 +73,7 @@ int ipvs_nl_send_message(struct nl_msg *
 {
 	int err = EINVAL;
 
-	sock = nl_handle_alloc();
+	sock = nl_socket_alloc();
 	if (!sock) {
 		nlmsg_free(msg);
 		return -1;
@@ -88,7 +88,7 @@ int ipvs_nl_send_message(struct nl_msg *
 
 	/* To test connections and set the family */
 	if (msg == NULL) {
-		nl_handle_destroy(sock);
+		nl_socket_free(sock);
 		sock = NULL;
 		return 0;
 	}
@@ -104,12 +104,12 @@ int ipvs_nl_send_message(struct nl_msg *
 
 	nlmsg_free(msg);
 
-	nl_handle_destroy(sock);
+	nl_socket_free(sock);
 
 	return 0;
 
 fail_genl:
-	nl_handle_destroy(sock);
+	nl_socket_free(sock);
 	sock = NULL;
 	nlmsg_free(msg);
 	errno = err;
diff -uprN a/Makefile b/Makefile
--- a/Makefile	2011-02-08 08:24:23.000000000 +0800
+++ b/Makefile	2017-03-31 15:12:01.899580751 +0800
@@ -37,10 +37,10 @@ RPMSPECDIR	= $(shell rpm --eval '%_specd
 
 CC		= gcc
 INCLUDE		=
-SBIN		= $(BUILD_ROOT)/sbin
+SBIN		= $(DESTDIR)/usr/sbin
 MANDIR		= usr/man
-MAN		= $(BUILD_ROOT)/$(MANDIR)/man8
-INIT		= $(BUILD_ROOT)/etc/rc.d/init.d
+MAN		= $(DESTDIR)/$(MANDIR)/man8
+INIT		= $(DESTDIR)/etc/init.d
 MKDIR		= mkdir
 INSTALL		= install
 STATIC_LIBS	= libipvs/libipvs.a
